scai-gen/pkg/generators/v1.go

```diff
@@ -0,0 +1,102 @@
+package generators
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/in-toto/scai-demos/scai-gen/pkg/policy"
+
+	ita "github.com/in-toto/attestation/go/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func NewRdForFile(filename string, name string, uri string, hashAlg string, withContent bool, mediaType string, downloadLocation string, annotations *structpb.Struct) (*ita.ResourceDescriptor, error) {
```

Comment:
nit: docstring

Reply:
also this is a long line, a small-delta way to cut this down would be to not repeat the parameter types for consecutive args of the same type.

```
func NewRdForFile(filename, name, uri, hashAlg string, withContent bool, mediaType, downloadLocation string, annotations *structpb.Struct) (*ita.ResourceDescriptor, error) {
```

If `withContent` is moved, you could get away with using a single `string` for all the params...

Reply:
Ah yes, sorry about missing the docstrings. Will add. Also forgot I could group params of the same type, thanks!